src/figuresClasses.ts

```diff
@@ -1,19 +1,92 @@
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
+
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
+
+  getArea(): number;
 
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Incorrect Data');
+    }
 
+    const longestSide = Math.max(a, b, c);
+
+    if (longestSide >= a + b + c - longestSide) {
+      throw new Error('Incorrect Data');
+    }
+  }
+
+  getArea(): number {
+    const {
+      a,
+      b,
+      c,
+    } = this;
+    const p = (a + b + c) / 2;
+
+    return Math.floor(Math.sqrt(p * (p - a) * (p - b) * (p - c)) * 100) / 100;
+  }
 }
 
-export class Circle {
+export class Circle implements Figure {
+  public shape: Shape = Shape.circle;
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Incorrect Data');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI * (this.radius ** 2) * 100) / 100;
```

Comment:
Create variable for area value
Also, remember about DRY. You are using next code at least twice, create function for this `Math.floor(... * 100) / 100;`